### PR TITLE
Pass id rather than object as the 5th argument to h5d_read

### PR DESCRIPTION
**Pull request by grero**
This should fix issue #140.

### CODE REVIEW

File: src/MAT_HDF5.jl [hunk 1]
```diff
@@ -622,7 +622,7 @@ function read(obj::HDF5Dataset, ::Type{Array{Bool}})
     end
     # Use the low-level HDF5 API to put the data directly into a Bool array
     tf = Array{Bool}(undef, size(obj))
-    HDF5.h5d_read(obj.id, HDF5.hdf5_type_id(UInt8), tf, obj.xfer)
+    HDF5.h5d_read(obj.id, HDF5.hdf5_type_id(UInt8), tf, obj.xfer.id)
```
Comment (jmert) [inline src/MAT_HDF5.jl:625]:
```suggestion
    HDF5.h5d_read(obj.id, HDF5.hdf5_type_id(UInt8), tf, convert(HDF5.Hid, obj.xfer))
```

Directly accessing `obj.xfer.id` will require bounding HDF5 to v0.13.2 or greater. The convert route preserves compatibility with older versions of HDF5.

**Comment by jmert** [comment]
CI failures are solved by #142.

**Comment by musm** [comment]
Does this need to also go in  0.8.1 ?

**Comment by jmert** [comment]
To avoid the deprecation warnings completely, yes, it should go in the next release. Thanks!

**Pull request merged**